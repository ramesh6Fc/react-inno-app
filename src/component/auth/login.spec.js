import React from 'react';
import { render, wait, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Login from './login';
import mockedAxios from 'axios';


describe('The login page', () => {
    it ('Log in renderer as expected',() => {
        const tree = renderer
                        .create(<Login />)
                        .toJSON();
                        console.log(tree, "login page elements");

                        expect(tree).toMatchSnapshot ();
    })
})
// import Service from "../../service";

// const axios = require('axios');
// jest.mock('axios');
// axios.get.mockResolvedValue({
//     data: []
// });

// // beforeEach(() => {
// //     render(<Login />);
// // });


// // import * as service from "../../service";

// // const iUsers = [{ message: "User Not found." }];
// // const iPwd = [ {
// //     accessToken: null,
// //     message: "Invalid Password!"
// //   }];
// // const user = [ {
// //     id: "8798",
// //     username: "abc",
// //     email: "abc@abc.com",
// //     accessToken: "token"
// //   } ];

// // const inValidUser = jest.spyOn(service, 'login');
// // mockGetUsers.mockResolvedValue(iUsers);

// // const inValidPwd = jest.spyOn(service, 'login');
// // mockGetRolesWithUsers.mockResolvedValue(iPwd);

// // const login = jest.spyOn(service, 'login');
// // mockGetRoles.mockResolvedValue(user);   

// afterEach(cleanup);
// test('mocking axios request', async () => {
//     const data = {
//     data: [ {username: "abc",
//         password: "Abcd123"}]
//     };

//     mockedAxios.post.mockResolvedValue(data);
//     const { getByText } = render(<App />);
//     await wait(() => {
//         console.log(getByText,"TTTTTTTTTTTTTTTTTTTTTTTT")
//     expect(getByText('mocked title'));
// });
// })



// describe('Test the login component', () => {
    
//     it('Test the login component with Invalid user and password', () => {
        
//     });

//     it('Test the login component with Invalid password', () => {
        
//     });

//     it('Test the successful login',async () => {
//         // var mock = new MockAdapter(axios);
//         // const data = { response: true };
//         // const gen = Service.login({username: 'abc', password :'Abcd123'})
//         // const { value } = await gen.next()
//        // expect(Service.login('abc', 'Abcd123')).toBe(3);
//     });
// });
