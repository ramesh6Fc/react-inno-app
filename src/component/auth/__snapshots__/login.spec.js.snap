// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The login page Log in renderer as expected 1`] = `
<div
  className="cards cards-container"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-elemt"
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <div>
        <input
          className="form-input"
          name="username"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-elemt"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <div>
        <input
          className="form-input"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="form-elemt"
    >
      <button
        className="button"
        disabled={false}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <span>
          Sign In
        </span>
      </button>
    </div>
    
    <button
      disabled={true}
      style={
        Object {
          "display": "none",
        }
      }
    />
  </form>
  <div
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "11px",
        "textAlign": "right",
      }
    }
  >
    <a
      href="!#"
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      Register
    </a>
  </div>
</div>
`;
